src/darwin/Framework/CHIP/MTRAsyncWorkQueue.mm

```diff
@@ -391,7 +391,7 @@ - (BOOL)hasDuplicateForTypeID:(NSUInteger)opaqueDuplicateTypeID workItemData:(id
             }
         }
     }
-    os_unfair_lock_unlock(&_lock);
+
     return hasDuplicate;
```

Comment:
Here we could get rid of `hasDuplicate` entirely and just have, inside the loop:
```
if (stop) {
    return isDuplicate;
}
```

and at the very end of the function:

```
return NO;
```

The only reason we had hasDuplicate was to carry around the state until we managed to unlock.

Reply:
Done